### PR TITLE
Separated method deleteMessages in MessagingEngine into two

### PR DESCRIPTION
**Pull request by sdkottegoda**
This PR separates the method deleteMessages in MessagingEngine as deleteMessages and moveToDLC since having a single method to perform both tasks was confusing.

Addresses the jira https://wso2.org/jira/browse/MB-1186

Please merge https://github.com/wso2/product-mb/pull/163 along with this or else PurgeMessagesTestCase might fail. Even though there is no direct connection, the test case started failing after this change.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java [hunk 1]
```diff
@@ -787,4 +784,33 @@ public long getLastAssignedSlotMessageId(String queueName) throws AndesException
...
+            AndesRemovableMetadataDTO dto = storageSeparatedAndesRemovableMetadataDTOs.get(message
+                    .getStorageDestination());
+            if (dto == null) {
+                dto = new AndesRemovableMetadataDTO();
+            }
+            if (message.getStorageDestination() != null) {
+                storageSeparatedAndesRemovableMetadataDTOs.put(message.getStorageDestination(), dto);
+            }
+            dto.messagesToRemove.add(message.getMessageID());
+            dto.msgCount = dto.msgCount + 1;
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java]: Can use dto.msgCount++ here(Increment).

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java [hunk 2]
```diff
@@ -787,4 +784,33 @@ public long getLastAssignedSlotMessageId(String queueName) throws AndesException
...
+                new HashMap<>(messageMetadataList.size());
+
+        for (AndesRemovableMetadata message : messageMetadataList) {
+            //update <storageQueue, dtos> map
+            AndesRemovableMetadataDTO dto = storageSeparatedAndesRemovableMetadataDTOs.get(message
+                    .getStorageDestination());
+            if (dto == null) {
+                dto = new AndesRemovableMetadataDTO();
+            }
+            if (message.getStorageDestination() != null) {
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java]: Flip here.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java [hunk 3]
```diff
@@ -787,4 +784,33 @@ public long getLastAssignedSlotMessageId(String queueName) throws AndesException
...
+            messageMetadataList) {
+
+        Map<String, AndesRemovableMetadataDTO> storageSeparatedAndesRemovableMetadataDTOs =
+                new HashMap<>(messageMetadataList.size());
+
+        for (AndesRemovableMetadata message : messageMetadataList) {
+            //update <storageQueue, dtos> map
+            AndesRemovableMetadataDTO dto = storageSeparatedAndesRemovableMetadataDTOs.get(message
+                    .getStorageDestination());
+            if (dto == null) {
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java]: Flip here

**Comment by ramith** [comment]
Please create a Jira.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 4]
```diff
@@ -1153,39 +1151,25 @@ public void deleteMessages(final String storageQueueName,
...
-                    metadataRemovalPreparedStatement.setLong(1, messageID);
-                    metadataRemovalPreparedStatement.addBatch();
+            metadataRemovalPreparedStatement = connection
+                    .prepareStatement(RDBMSConstants.PS_DELETE_METADATA);
 
-                }
-                metadataRemovalPreparedStatement.executeBatch();
+            for (Long messageID : messagesToRemove) {
+                //add parameters to delete metadata
+                metadataRemovalPreparedStatement.setLong(1, messageID);
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: 1L?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java [hunk 5]
```diff
@@ -787,4 +784,33 @@ public long getLastAssignedSlotMessageId(String queueName) throws AndesException
...
+            AndesRemovableMetadataDTO dto = storageSeparatedAndesRemovableMetadataDTOs.get(message
+                    .getStorageDestination());
+            if (null == dto) {
+                dto = new AndesRemovableMetadataDTO();
+            }
+            if (null != message.getStorageDestination()) {
+                storageSeparatedAndesRemovableMetadataDTOs.put(message.getStorageDestination(), dto);
+            }
+            dto.messagesToRemove.add(message.getMessageID());
+            dto.msgCount = dto.msgCount++;
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java]: Can remove assignment.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/CQLBasedMessageStoreImpl.java [hunk 6]
```diff
@@ -800,26 +799,16 @@ public void deleteMessages(final String storageQueueName, List<Long> messagesToR
...
-                        where(eq(CQLConstants.QUEUE_NAME, storageQueueName)).
-                        setConsistencyLevel(config.getWriteConsistencyLevel()));
-                for (Long messageID : messagesToRemove) {
-                    batchStatement.add(psDeleteMessagePart.bind(messageID));
-                }
+
+            for (Long messageID : messagesToRemove) {
+                batchStatement.add(psDeleteMetadata.bind(
+                        storageQueueName,
+                        messageID
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/CQLBasedMessageStoreImpl.java]: Wrap this to one line.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/HectorBasedMessageStoreImpl.java [hunk 7]
```diff
@@ -624,23 +622,13 @@ public void deleteMessages(final String storageQueueName,
...
-                    rows2Remove.add(MESSAGE_CONTENT_CASSANDRA_ROW_NAME_PREFIX +
-                            messageID.longValue());
-                }
+            for (Long messageID : messagesToRemove) {
+                HectorDataAccessHelper
+                        .deleteLongColumnFromRaw(
+                                HectorConstants.META_DATA_COLUMN_FAMILY,
+                                storageQueueName, messageID.longValue(), mutator, false);
+                rows2Remove.add(MESSAGE_CONTENT_CASSANDRA_ROW_NAME_PREFIX +
+                                messageID.longValue());
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/HectorBasedMessageStoreImpl.java]: Wrap this to one line.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 8]
```diff
@@ -1153,39 +1151,25 @@ public void deleteMessages(final String storageQueueName,
...
-            if (!deleteAllMetaData) {
-                metadataRemovalPreparedStatement = connection
-                        .prepareStatement(RDBMSConstants.PS_DELETE_METADATA);
-
-                for (Long messageID : messagesToRemove) {
-                    //add parameters to delete metadata
-                    metadataRemovalPreparedStatement.setLong(1, messageID);
-                    metadataRemovalPreparedStatement.addBatch();
+            metadataRemovalPreparedStatement = connection
+                    .prepareStatement(RDBMSConstants.PS_DELETE_METADATA);
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: Can we make this one line?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java (hunk 4 above)
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: It sets the parameter number. Therefore, it needs to be an int.

**Pull request merged**